### ISSUE TITLE
libwebp requires turbo_jpeg

### ISSUE THREAD

**Issue by kallisti5** [haikuports/haikuports#4292]
```
problem 1: nothing provides lib:libjpeg>=62.3.0 needed by libwebp-1.0.3-1
```

libwebp (a base build package) requires libjpeg_turbo vs jpeg due to the version number.
We ship standard jpeg in the images.

We *could* switch jpeg to libjpeg_turbo in our images, however libjpeg_turbo is flagged as unstable on all architectures except x86_64.

Example:

```diff
diff --git a/build/jam/BuildFeatures b/build/jam/BuildFeatures
index 47bd6ece84..29ec1d7efb 100644
--- a/build/jam/BuildFeatures
+++ b/build/jam/BuildFeatures
@@ -440,11 +440,11 @@ if [ IsPackageAvailable jasper_devel ] {
 
 
 # jpeg
-if [ IsPackageAvailable jpeg_devel ] {
+if [ IsPackageAvailable libjpeg_turbo_devel ] {
        ExtractBuildFeatureArchives jpeg :
-               file: base jpeg
+               file: base libjpeg_turbo
                        runtime: lib
-               file: devel jpeg_devel
+               file: devel libjpeg_turbo_devel
                        depends: base
                        library: $(developLibDir)/libjpeg.so
                        headers: $(developHeadersDir)
```

Options:
  * Change the libwebp dependency on libjpeg_turbo to the normal jpeg we ship
  * Update libjpeg_turbo to be available on all architectures (at minimum x86_64, x86_gcc2... we could work on "fixing" the rest later.)

@korli , @pulkomandy   thoughts?

**Comment by pulkomandy** [comment]
libjpeg-turbo should (if configured properly) provide ABI compatibility with libjpeg 6, I think 7, and 8. But it does not provide ABI compatibility with libjpeg 9.

I don't see why it is disabled on other platforms, I guess just lack of testing?

It does also not provide the new image formats in libjpeg 9, but I don't know if that's useful (according to libjpeg-turbo statements, they are nonstandard extensions, so I would not allow using them for encoding, but if we ever come around a JPEG file using them, it would be nice to still be able to decode.

**Comment by kallisti5** [comment]
yeesh. That's kind of a mess.

My Fedora 30 desktop ships libjpeg-turbo:
```
$ rpm -qa | grep jpeg
openjpeg2-2.3.1-1.fc30.x86_64
libjpeg-turbo-devel-2.0.2-1.fc30.i686
libjpeg-turbo-devel-2.0.2-1.fc30.x86_64
libjpeg-turbo-2.0.2-1.fc30.i686
libjpeg-turbo-2.0.2-1.fc30.x86_64
```

Ubuntu Bionic seems to ship both:
```
$ podman run -it ubuntu:latest /bin/bash -l
root@92f95861ee37:/# dpkg -l | grep jpeg
root@92f95861ee37:/# apt-get install firefox
.
.
root@92f95861ee37:/# dpkg -l | grep jpeg
ii  libjpeg-turbo8:amd64           1.5.2-0ubuntu5.18.04.1          amd64        IJG JPEG compliant runtime library.
ii  libjpeg8:amd64                 8c-2ubuntu8                     amd64        Independent JPEG Group's JPEG runtime library (dependency package)
```


Wouldn't changing from libjpeg to libjpeg-turbo cause some package conflicts for upgrading users?

**Comment by pulkomandy** [comment]
A pkgman full-sync would make the upgrade go through.

**Comment by kallisti5** [comment]
actually.. I was talking with waddlesplash on this, the conflicts are already setup properly.  It seems you can have jpeg and libjpeg_turbo installed at the same time.

The conflict is on libjpeg_turbo_devel and jpeg_devel (which makes sense, the libraries have different so versioning, but conflicting headers)

I saw @korli just made this commit.. 
https://github.com/haikuports/haikuports/commit/20131b657352f65e0ba00417d6437e4b9988e789

Not 100% sure what the reasoning is yet.


I just built libjpeg_turbo on x86_gcc2 successfully... updating the arches.

**Comment by kallisti5** [comment]
seems like this was actually solved via 20131b657352f65e0ba00417d6437e4b9988e789    A new libwebp means it no longer depends on libturbo_jpeg.
